### PR TITLE
Add airtap for continuous browser tests

### PR DESCRIPTION
**Pull request by vweevers**
Made a few quick hacks to get most tests passing in local Chrome at least.

What's the concurrency limit of the Sauce Labs account? You may want to change Travis's concurrency setting depending on that.

### CODE REVIEW

File: .airtap.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+browsers:
+  - name: ie
+    version: 11..latest
+  - name: opera
+    version: -3..latest
+  - name: chrome
+    version: -3..latest
+  - name: firefox
+    version: -3..latest
+  # - name: safari
```
[the remark this answers is not in the document]
Comment (mcollina) [reply .airtap.yml]: if they are passing, let's go for it.

**Comment by mcollina** [comment]
> What's the concurrency limit of the Sauce Labs account? You may want to change Travis's concurrency setting depending on that.

I would like to ask them to bump it, after this lands.

File: .airtap.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vweevers) [reply .airtap.yml]: We won't know until you merge ;) I didn't run the tests on my own SL account or anything.

I'll uncomment safari and we'll see.

**Comment by mcollina** [comment]
Can you please allow me to push to this branch? I've made some changes in the next branch, and I would like to push a rebase.

**Comment by mcollina** [comment]
I'm currently getting:

```
$ NODE_OPTIONS="--trace-warnings" npm run test-browsers

> readable-stream@3.0.0-dev0 test-browsers /Users/matteo/Repositories/readable-stream
> airtap --sauce-connect --loopback airtap.local --no-coverage test/browser.js

- testing: chrome @ Windows 10: 67
- queuing: <chrome 67 on Windows 10>
- restarting: <chrome 67 on Windows 10>
- restarting: <chrome 67 on Windows 10>
- restarting: <chrome 67 on Windows 10>
- restarting: <chrome 67 on Windows 10>
- restarting: <chrome 67 on Windows 10>
- restarting: <chrome 67 on Windows 10>
(node:72841) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 done listeners added. Use emitter.setMaxListeners() to increase limit
    at _addListener (events.js:280:19)
    at SauceBrowser.addListener (events.js:297:10)
    at SauceBrowser.once (events.js:341:8)
    at /Users/matteo/Repositories/readable-stream/node_modules/airtap/bin/airtap.js:271:21
    at next (/Users/matteo/Repositories/readable-stream/node_modules/batch/index.js:129:7)
    at Batch.end (/Users/matteo/Repositories/readable-stream/node_modules/batch/index.js:169:5)
    at shutdownAllBrowsers (/Users/matteo/Repositories/readable-stream/node_modules/airtap/bin/airtap.js:275:15)
    at Zuul.<anonymous> (/Users/matteo/Repositories/readable-stream/node_modules/airtap/bin/airtap.js:245:9)
    at emitOne (events.js:116:13)
    at Zuul.emit (events.js:211:7)
```

When running things locally.

**Comment by vweevers** [comment]
@ralphtheninja mind helping out with the airtap error above? I'm short on time

**Comment by mcollina** [comment]
Also on Travis we are getting: "Skipping Airtap due to no secure Travis environment"  Why is that happening?

**Comment by vweevers** [comment]
@mcollina Travis secure environment variables are not available in PRs from external forks. Airtap doesn't run in this case because it would error (due to missing Sauce Labs credentials).

**Comment by mcollina** [comment]
Oh ok. So I should really just test it locally and then push to a new branch.
Have you got any idea it does not work locally for me?

**Comment by vweevers** [comment]
> Have you got any idea it does not work locally for me?

By locally do you mean in a local browser, or Sauce Labs? Because Airtap does not support the latter (yet), unless you manually run a Sauce Connect tunnel.

File: package.json [hunk 2]
```diff
@@ -24,6 +25,8 @@
   "scripts": {
     "test": "tap -j 4 test/parallel/*.js test/ours/*.js",
     "ci": "TAP=1 tap test/parallel/*.js test/ours/*.js | tee test.tap",
+    "test-browsers": "airtap --sauce-connect -- test/browser.js",
+    "test-browser-local": "airtap --local --open -- test/browser.js",
```
Comment (ralphtheninja) [inline package.json]: :+1:

File: .travis.yml [hunk 3]
```diff
@@ -22,5 +22,5 @@ matrix:
 script: npm run $CMD
 env:
   global:
-  - secure: nk/kDUZV385ey9kkrGi6EHvnslTAW+GEHGrjbFIsb8Foq01bL9oj0uPaVc5kaVWCHst3WsIdVMA1Hos/lN/WMHRyfOiSrVVWv0sVnzg0YaM9cBbn5jwXn30Z645REgQcoEDKrYFp8rE3RpPFgYDFWNh948qwPtwtnwNhoq/Gd9A=
-  - secure: faODkEUSlbH4fLBll4UkHeSEsCZdW+Hw+Sb6yDX5PXkIgq5wvqALIUgDIEBz2F8gciLlPlIssR2csKPqYIK5uwUq7uPBJVmwcf6i0K9NZaU/WaGaUVCzs2oMAoN007pTU30USEiWzfEGR+0d5hRKoeJdZIU0nBxkpLY4ZWM0SGU=
+  - secure: MMYiNZ8ICzNkz1OQ3/jFrcBVxG5eIGoUn1/IRL9JeWbB6gWg8kxAMFXZmLDJrJj5Zm6VD31mpX6PS3Ot46syMl+FnnyNCdqcCJUuUYCdRS8PanUnVq2QL1+9f2uT/S3UeQJhTaFXAzHA4cSNcaXlSB0QzbiDi0xZUobAAM/h5sw=
```
Comment (ralphtheninja) [inline .travis.yml]: @mcollina This worked, right?

Comment (mcollina) [reply .travis.yml]: No it was skipped.

> Skipping Airtap due to no secure Travis environment.

**Pull request closed without merging**